src/GitVersionExe/GitVersionApp.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace GitVersion
+{
+    internal class GitVersionApp : IHostedService
+    {
+        private readonly IHostApplicationLifetime applicationLifetime;
+        private readonly IGitVersionRunner gitVersionRunner;
+        private readonly Arguments arguments;
+
+        public GitVersionApp(IHostApplicationLifetime applicationLifetime, IGitVersionRunner gitVersionRunner, IOptions<Arguments> options)
+        {
+            arguments = options.Value;
+            this.applicationLifetime = applicationLifetime;
+            this.gitVersionRunner = gitVersionRunner;
+        }
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                gitVersionRunner.Run(arguments);
```

Comment:
If `this.` is being used in one place, shouldn't it be everywhere?

Reply:
Line 18 and 19 have this because the parameter name is the same as the field name, as for the arguments, there is no arguments parameter. I can change that to prefix it with this

Reply:
I agree that `this.` should be used consistently. 👍

Reply:
fixed